vignettes/adrs_imwg.Rmd

```diff
@@ -0,0 +1,897 @@
+parameters (biochemical analyses). These need to be two distinct samples and 
+usually is performed on the same day or at next scheduled visit.  
+Bone marrow assessments and imaging do not need to be confirmed. 
+
+If `RS` contains unconfirmed response and confirmation is performed at 
+next scheduled visit we can derive Confirmed Overall Response based on 
+response at subsequent visit.
+
+While deriving confirmed response, there are some rules to follow:
+
+* Until any response category has been confirmed, response = stable disease (`SD`).  
+* Patients will continue in the last confirmed response category until 
+there is confirmation of progression or improvement to a higher response status.  
+* If the result is not confirmed by a second evaluation, the status is `NE`
+and the prior confirmed disease status remains valid.  
+* Once confirmed: date of the initial test is considered as date of response.
+
+Below table provides a summary of the confirmed response status calculation at 
+each time point.
+
+```{r echo=FALSE}
+list_resp <- tribble(
+  ~"Response 1 time point",   ~"Response 2 time point",
+  ~"Confirmed response 1 time point",
+  "sCR", "sCR", "sCR",
+  "CR", "sCR/CR", "max (CR, last confirmed status)",
+  "VGPR", "sCR/CR/VGPR", "max(VGPR, last confirmed status)",
+  "PR", "sCR/CR/VGPR/PR", "max(PR, last confirmed status)",
+  "MR", "sCR/CR/VGPR/PR/MR", "max(MR, last confirmed status)",
+  "sCR", "CR", "last confirmed status",
+  "sCR/CR", "VGPR", "last confirmed status",
+  "sCR/CR/VGPR", "PR", "last confirmed status", 
+  "sCR/CR/VGPR/PR", "MR", "last confirmed status",
+  "sCR/CR/VGPR/PR/MR", "SD/PD/NE/NA", "last confirmed status",
+  "SD", "any", "last confirmed status",
+  "NE", "any", "last confirmed status",
+  "PD reason imaging", "any", "PD",
+  "PD reason serum/urine", "PD/death", "PD",
+  "PD reason serum/urine", "sCR/CR/VGPR/PR/MR/SD/NE/NA", "last confirmed status",
+)
+
+knitr::kable(list_resp)
+```
+
+At the next step we join records with subsequent responses. If response requires confirmation and consecutive assessment does not confirm previous results, we reassign response to `NE`.
+
+```{r}
+adrs_joined <- 
+  derive_vars_joined(
+    adrs,
+    adrs,
+    by_vars = exprs(USUBJID),
+    order = exprs(ADY),
+    new_vars = exprs(AVALC.next = AVALC, AVAL.next = AVAL),
+    tmp_obs_nr_var = ASEQ,
+    join_type = "after",
+    filter_join = ASEQ + 1 == ASEQ.join,
+    mode = "first",
+  )  %>%
+  mutate(AVALC.confirmed = case_when(
+    #better response
+    AVALC %in% c("sCR", "CR", "VGPR", "PR", "MR") & 
+      AVALC.next %in% c("sCR", "CR", "VGPR", "PR", "MR") & 
+      AVAL.next >= AVAL ~ AVALC,
+    #worse response
+    AVALC %in% c("sCR", "CR", "VGPR", "PR", "MR") & 
+      AVALC.next %in% c("sCR", "CR", "VGPR", "PR", "MR", "SD", "PD") & 
+      AVAL.next < AVAL ~ "NE",
+    #next assessment NE or NA
+    AVALC %in% c("sCR", "CR", "VGPR", "PR", "MR") & 
+      (AVALC.next == "NE" | is.na(AVALC.next)) ~ "NE",
```

Comment:
In this step we would assign NE but in next step it will change for SD. "Until any response is confirmed, we need to assign a value of `SD`. " @amitjaingsk please comment on this

Reply:
@starosto @bundfussr little confused with the intermediate steps. Based on the work flow row#10 it should be SD provided previously response is not confirmed. Also based on the output results doesn;t look correct to me. for eg 701-1028 subject, the sCR is confirmed on Wk 6 visit, so we should continue that response until either we have a better confirmed response. So , NE should on wk12,wk18 should be sCR.

Reply:
updated